monitor/services/src/main/java/org/datacleaner/monitor/server/components/ComponentHandler.java

```diff
@@ -0,0 +1,251 @@
+
+    private final String componentName;
+    private DataCleanerConfiguration dcConfiguration;
+
+    ComponentDescriptor descriptor;
+    Map<String, MutableColumn> columns;
+    Map<String, InputColumn> inputColumns;
+    MyMutableTable table;
+    Component component;
+
+    public ComponentHandler(DataCleanerConfiguration dcConfiguration, String componentName) {
+        this.dcConfiguration = dcConfiguration;
+        this.componentName = componentName;
+    }
+
+    public void createComponent(ComponentConfiguration componentConfiguration) {
+        columns = new HashMap<>();
+        inputColumns = new HashMap<>();
+        descriptor = dcConfiguration.getEnvironment().getDescriptorProvider().getTransformerDescriptorByDisplayName(componentName);
+        table = new MyMutableTable("inputData");
+        if(descriptor == null) {
+            descriptor = dcConfiguration.getEnvironment().getDescriptorProvider().getAnalyzerDescriptorByDisplayName(componentName);
+        }
+        if(descriptor == null) {
+            throw ComponentNotFoundException.createTypeNotFound(componentName);
+        }
+        component = (Component) descriptor.newInstance();
+
+
+        // create "table" according to the columns specification (for now only a list of names)
+        int index = 0;
+        for (String columnName : componentConfiguration.getColumns()) {
+            MutableColumn column = columns.get(columnName);
+            if(column == null) {
+                column = new MutableColumn(columnName, ColumnType.VARCHAR, table, index, true);
+                columns.put(columnName, column);
+            } else {
+                column.setColumnNumber(index);
+            }
+            table.addColumn(index, column);
+            index++;
+        }
+
+        // Set the configured properties
+
+        for(String propertyName: componentConfiguration.getPropertiesNames()) {
+            ConfiguredPropertyDescriptor propDesc = descriptor.getConfiguredProperty(propertyName);
+            if(propDesc == null) {
+                LOGGER.debug("Unknown configuration property '" + propertyName + "'");
+                continue;
+            }
+            JsonNode userPropValue = componentConfiguration.getProperty(propDesc.getName());
+            if(userPropValue != null) {
+                if(propDesc.isInputColumn()) {
+                    List<String> colNames = convertToStringArray(userPropValue);
+                    List<InputColumn> inputCols = new ArrayList<>();
+                    for(String columnName: colNames) {
+                        inputCols.add(getOrCreateInputColumn(columnName));
+                    }
+                    propDesc.setValue(component, inputCols.toArray(new InputColumn[inputCols.size()]));
+                } else {
+                    Object value = convertPropertyValue(propDesc, userPropValue);
+                    propDesc.setValue(component, value);
```

Comment:
Here I would prefer to create a ComponentConfiguration object and set the values on that. Then below you would call LifeCycleHandler.assignConfiguredProperties(...)